packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/framework.ts

```diff
@@ -13,6 +13,14 @@ export = {
   [consts.FRAMEWORK_ON_TIMEOUT_HANDLER_NAME]: onTimeout,
 };
 
+const DEFAULT_DELAY = 10_000;
+const MAX_TOTAL_DELAY = 620_000;
```

Comment:
Just curious how you came to a little over ten minutes a the desired total delay.

Reply:
lambdas run for 15, and I figured the function could conceivably do anything after this code runs, so it makes sense to have some headroom; 5 minutes seemed to be a decent amount of time. Not the most rigorous process, this can be changed if needed.